### ISSUE TITLE
[import/newline-after-import] false positive with typescript import = require

### ISSUE THREAD

**Issue by ext** [benmosher/eslint-plugin-import#1654]
When using `import/newline-after-import` with typescript [`import = require(..)`](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require) I get a false positive requiring me to put a newline in the middle of the imports.

```typescript
import foo from "foo";
import bar = require("bar");
import baz from "baz";
```

## Expected results

No errors

## Actual results

>  1:1   error    Expected 1 empty line after import statement not followed by another import  import/newline-after-import

**Comment by armano2** [comment]
note: import equal declaration uses "special" syntax and and has typescript specific node type, [`TSImportEqualsDeclaration`](https://astexplorer.net/#/gist/93cb4f8113c0a391ea11ec9095f0a8e0/04d2941d057ebd9eabeab815b8079495f19443bf)

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Comment by ljharb** [comment]
While that syntax is only required/useful when TS's module system is broken (ie, when synthetic imports and esModuleInterop are not both enabled), I agree that code using it should work with this plugin.

**Comment by sidoruk-sv** [comment]
Also got into valid for typescript case:
```
import {API} from './some/exports';
import InnerType = API.some.comfort.namespaces.InnerType;
```

**Comment by patrickhulce** [comment]
More broadly, is there any option to get the plugin to always treat `import X = require('Y')` as though they were regular `import` statements? 

This seems to have broader effects in other rules as well. `--fix` won't fix sort order for anything that comes after the first `= require` statement for example. This unfortunately halted our migration from tslint since it was a show stopper for us :/

It's marked as help wanted if anyone could point me in the right direction I'd be happy to give it a shot :)

**Comment by ljharb** [comment]
(I'd still strongly recommend everyone migrate away from TS-specific syntax; `import X from 'Y'` is what should be used in that case, with synthetic imports and esModuleInterop enabled, as `tsc --init` enables)
